jackson-apt-processor/pom.xml

```diff
@@ -105,6 +105,22 @@
                 <groupId>org.sonatype.plugins</groupId>
                 <artifactId>nexus-staging-maven-plugin</artifactId>
             </plugin>
+            <plugin>
```

Comment:
I would like to have 2 versions, shaded and non shaded jars

Reply:
Well, this is a good question indeed. I suggest to use the shaded jar as a main artifact. 

In order to keep classpath clean, the prefered way for running apt should be by using the "-processorpath" command line argument. If the apt has external dependencies, we have to pass them as well. Using self contained (or uber) jar for the apt looks the correct way for me. Please, share your thoughts on this topic.

Reply:
I think that even if the uber jar is not the primary artifact we still can pass it in the command line to avoid passing the external dependencies, so i am with having a shaded jar version, but as a secondary artifact as an option, but for most of common cases where we are using it from maven or IDE i prefer to stick with a convention for artifact names, for example the shade plugin documentation suggest that we publish the shaded artifact as the secondary one

https://maven.apache.org/plugins/maven-shade-plugin/examples/attached-artifact.html

What i mean is that i expect when i am using this lib i will find the original jar under the name jackson-apt-processor and not under original-jackson-apt-processor, but if i am looking for a shaded version i can expect it to be jackson-apt-processor-fat or what ever post-fix.

so i am suggesting that we just add a classifier for the shaded jar and leave the original jar name as is.

Reply:
@vegegoku That makes sense! Let's do it like this. I will use "-uber" post fix for shaded jar.

Reply:
I'm reviewing this in the context of trying to follow some best practices (see https://github.com/google/auto/tree/master/common#processor-resilience). First, I want to understand the work already done here, why this ended up done this way. 

My preference would be for the "original" to be hidden away, unless someone really wants it for some reason, and for the default to be the shaded/uber jar, with all dependencies included - I think this is the original purpose of the patch, if I'm reading these comments correctly. The chief change I want to make from the original patch is to "relocate" external dependencies, so that in case this processor depends on version 1 of jar X, and another processor depends on version 2 of jar X, they won't conflict and result in compile failures.

My interpretation of @vegegoku's remarks leads me to think he may have mis-interpreted the `attached-artifact.html` link above:

> one can configure the plugin to attach the shaded artifact as a secondary artifact:

To me at least, this does not say "you should do this", but "it is possible to do this". Also, this does not make a different artifactId (though it certainly could do so if you added a second pom), but only adds a classifier (which appends the suffix after the version, not at the end of the artifact name).

Thoughts?

Reply:
The original idea was to have the uber jar as default and deal with cleaner classpath during application compilation. 
To me, "self-contained" package for a typical annotation processor sounds the wright way to go.